internal/pkg/resourceconfig/resourceconfig.go

```diff
@@ -135,29 +134,51 @@ func (p *RawConfigFileProvider) IsSupported(path string) bool {
 }
 
 // GetConfig returns the resource configs
-func (p *RawConfigFileProvider) GetConfig(path string) ([]*unstructured.Unstructured, error) {
+// from a directory or a file containing raw Kubernetes resource configurations
+func (p *RawConfigFileProvider) GetConfig(root string) ([]*unstructured.Unstructured, error) {
 	var values clik8s.ResourceConfigs
 
-	b, err := ioutil.ReadFile(path)
-	if err != nil {
-		return nil, err
-	}
-	objs := strings.Split(string(b), "---")
-	for _, o := range objs {
-		body := map[string]interface{}{}
+	err := filepath.Walk(root, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 
-		if err := yaml.Unmarshal([]byte(o), &body); err != nil {
-			return nil, err
+		if fi.IsDir() {
+			return nil
 		}
-		values = append(values, &unstructured.Unstructured{Object: body})
-	}
 
+		b, err := ioutil.ReadFile(path)
+
+		if err != nil {
+			return err
+		}
+		objs := strings.Split(string(b), "---")
```

Comment:
yaml decoder from sigs.k8s.io uses go-yaml underneath.
go-yaml supports multiple docs as explained in this pull req:
https://github.com/go-yaml/yaml/pull/301/files

It would return an array of interfaces{}

Reply:
The yaml decoder each time can only load a single resource.
For a multi resource yaml file, we need to use a for loop.
```
func (s *S) TestDecoder(c *C) {
	for i, item := range decoderTests {
		c.Logf("test %d: %q", i, item.data)
		var values []interface{}
		dec := yaml.NewDecoder(strings.NewReader(item.data))
		for {
			var value interface{}
			err := dec.Decode(&value)
			if err == io.EOF {
				break
			}
			c.Assert(err, IsNil)
			values = append(values, value)
				}
		c.Assert(values, DeepEquals, item.values)
}
```